protocol/nats_jetstream/v3/message_test.go

```diff
@@ -190,3 +198,116 @@ func TestGetExtension(t *testing.T) {
 		})
 	}
 }
+
+func TestFinish(t *testing.T) {
+	type args struct {
+		err    error
+		ackErr error
+		nakErr error
+	}
+	type wants struct {
+		err        error
+		ackCalled  bool
+		nackCalled bool
+	}
+	tests := []struct {
+		name  string
+		args  args
+		wants wants
+	}{
+		{
+			name: "nil error given",
+			args: args{
+				err: nil,
+			},
+			wants: wants{
+				err:        nil,
+				ackCalled:  true,
+				nackCalled: false,
+			},
+		},
+		{
+			name: "ACK error given",
+			args: args{
+				err: protocol.ResultACK,
+			},
+			wants: wants{
+				err:        nil,
+				ackCalled:  true,
+				nackCalled: false,
+			},
+		},
+		{
+			name: "NACK error given",
+			args: args{
+				err: protocol.ResultNACK,
+			},
+			wants: wants{
+				err:        nil,
+				ackCalled:  false,
+				nackCalled: true,
+			},
+		},
+		{
+			name: "unknown error given",
+			args: args{
+				err: errors.New("unknown"),
+			},
+			wants: wants{
+				err:        nil,
```

Comment:
So we're slurping the error and returning `nil`? Is this the correct behavior when finishing?

Reply:
This is the previous behavior.  I am open to changing this behavior if there is support to change it.

Reply:
Got it, good question if we want consistency (but ambiguous behavior) or fix it. What do you think? Perhaps someone using `v3` would not use the prior implementations and therefore we should do what's right for users?

Reply:
After looking at 
https://github.com/cloudevents/sdk-go/blob/main/v2/client/client.go#L263
and
https://github.com/cloudevents/sdk-go/blob/main/v2/client/invoker.go#L59

I believe the error returned from Finish is only used for logging:
`cecontext.LoggerFrom(ctx).Warn("Error while handling a message: ", err)`

So the real question is more about whether we should call Ack() or Nak() in the unknown error case.
In the case that Ack() or Nak() is not called, the ACK/NACK behavior on the server is based on how the NATS consumer is setup.
There are three options: AckNonePolicy, AckAllPolicy, AckExplicitPolicy

So I would think if you had AckNonePolicy or AckAllPolicy, not calling Ack() would result in the server auto-acking.
If you had AckExplicitPolicy set, then failing to call Ack() would result in the message being redelivered.

This might be correct as it would "default" to how the user setup the consumer.  
NATS has a MaxDeliver option which would give up after the sending "MaxDeliver times", but the default is "unlimited".

I am inclined to give the user the benefit of defining that behavior, but I am not married to that feeling.

Reply:
Yeah, that sounds reasonable! Feel free to document the behavior e.g., adding a README or code comments.

Reply:
Added a comment.